rust/lance/src/format/index.rs

```diff
@@ -44,6 +46,42 @@ pub struct Index {
     pub fragment_bitmap: Option<RoaringBitmap>,
 }
 
+impl Index {
+    /// Returns the fragment ids that are not indexed by this index.
+    pub async fn unindexed_fragments(&self, dataset: &Dataset) -> Result<Vec<Fragment>> {
```

Comment:
nit: wanna expose this to python too?

Reply:
Also, side note: In the future, we should probably avoid returning types that allocate if we can instead return `impl Iterator<>`. The caller may not need to collect the data, or if they do, they might need it in a different data structure (HashSet, BTreeMap) or just need to collect one field (the fragment ids).